liquid-generator.js

```diff
@@ -0,0 +1,21 @@
+var xx = "water"
+var yy = "sand-water"
+var zz = "darksand-water"
+var cc = "darksand-tainted-water"
+const liquidGenerator = extendContent(SolarGenerator,"liquidGenerator",{
+getPowerProduction(tile){
+    if(tile.floor() == xx || tile.floor() == yy || tile.floor() == zz || tile.floor() == cc){
+        return 1.4
```

Comment:
instead of using the alphabet you should probably use liquid drop / affinities ^ (though way more is wrong here*)